contracts/contracts/coordination/ManagedAllowList.sol

```diff
@@ -2,149 +2,97 @@
-        subscription = _subscription;
+    constructor(Coordinator _coordinator) GlobalAllowList(_coordinator) {
+        _disableInitializers();
     }
 
-    /**
-     * @notice Checks if the sender is the authority of the ritual
-     * @param ritualId The ID of the ritual
-     */
-    modifier onlyCohortAuthority(uint32 ritualId) {
-        require(
-            coordinator.getAuthority(ritualId) == msg.sender,
-            "Only cohort authority is permitted"
-        );
-        _;
-    }
-
-    /**
-     * @notice Checks if the sender has allowance to set authorizations
-     * @dev This function overrides the canSetAuthorizations modifier in the GlobalAllowList contract
-     * @param ritualId The ID of the ritual
-     */
-    modifier canSetAuthorizations(uint32 ritualId) override {
-        require(getAllowance(ritualId, msg.sender) > 0, "Only administrator is permitted");
-        _;
+    function ritualRole(uint32 ritualId, bytes32 role) public view returns (bytes32) {
+        return keccak256(abi.encodePacked(ritualId, role));
     }
 
     /**
-     * @notice Returns the allowance of an administrator for a ritual
+     * Acquire cohort admin role
      * @param ritualId The ID of the ritual
-     * @param admin The address of the administrator
-     * @return The allowance of the administrator
      */
-    function getAllowance(uint32 ritualId, address admin) public view returns (uint256) {
-        return allowance[LookupKey.lookupKey(ritualId, admin)];
+    function initializeCohortAdminRole(uint32 ritualId) public {
+        bytes32 cohortAdminRole = ritualRole(ritualId, COHORT_ADMIN_BASE);
+        address authority = coordinator.getAuthority(ritualId);
+        require(authority != address(0), "Ritual is not initiated");
+        if (hasRole(cohortAdminRole, authority)) {
+            return;
+        }
+        _setRoleAdmin(ritualRole(ritualId, AUTH_ADMIN_BASE), cohortAdminRole);
+        _grantRole(cohortAdminRole, authority);
     }
 
-    /**
-     * @dev This function is called before the setAuthorizations function
-     * @param ritualId The ID of the ritual
-     * @param addresses The addresses to be authorized
-     * @param value The authorization status
-     */
-    function _beforeSetAuthorization(
-        uint32 ritualId,
-        address[] calldata addresses,
-        // TODO: Currently unused, remove?
-        bool value
-    ) internal override {
-        super._beforeSetAuthorization(ritualId, addresses, value);
-        for (uint256 i = 0; i < addresses.length; i++) {
-            require(
-                authActions[ritualId] <
-                    subscription.authorizationActionsCap(ritualId, addresses[i]),
-                "Authorization cap exceeded"
-            );
-        }
+    function grantAuthAdminRole(uint32 ritualId, address account) external {
+        initializeCohortAdminRole(ritualId);
+        grantRole(ritualRole(ritualId, AUTH_ADMIN_BASE), account);
```

Comment:
So just to confirm, if the ritual authority calls `grantRole()` directly without having first called `initializeCohortAdminRole()`, the call will fail.

But, if the ritual authority either:
- makes the two separate calls in here themselves directly
OR 
- calls `grantAuthAdminRole()`

then granting the role will work correctly.